src/index.js

```diff
@@ -177,48 +180,101 @@ class Saturn {
   }
 
   async _reportLogs () {
-    if (this.logs.length) {
-      await fetch(
-        'https://twb3qukm2i654i3tnvx36char40aymqq.lambda-url.us-west-2.on.aws/',
-        {
-          method: 'POST',
-          body: JSON.stringify({
-            bandwidthLogs: this.logs
-          })
-        }
-      )
-      this.logs = []
+    if (!this.logs.length) {
+      return
+    }
+
+    let bandwidthLogs = this.logs
+    if (this.hasPerformanceAPI) {
+      bandwidthLogs = this._matchLogsWithPerformanceMetrics(bandwidthLogs)
     }
+
+    await fetch(
+      this.opts.logURL,
+      {
+        method: 'POST',
+        body: JSON.stringify({ bandwidthLogs, logSender: this.opts.logSender })
+      }
+    )
+
+    this.logs = []
+    this._clearPerformanceBuffer()
+  }
+
+  /**
+   *
+   * @param {Array<object>} logs
+   */
+  _matchLogsWithPerformanceMetrics (logs) {
+    const logsToReport = []
+
+    for (const log of logs) {
+      const metrics = this._getPerformanceMetricsForLog(log)
+
+      if (!metrics.isFromBrowserCache) {
+        delete metrics.isFromBrowserCache
+        Object.assign(log, metrics)
+        logsToReport.push(log)
+      }
+    }
+
+    return logsToReport
   }
 
   /**
    *
    * @param {object} log
+   * @returns {object}
    */
-  _addPerformanceAPIMetrics (log) {
-    if (typeof window !== 'undefined' && window?.performance) {
-      const entry = performance
-        .getEntriesByType('resource')
-        .find((r) => r.name === log.url.href)
-      if (entry) {
-        const dnsStart = entry.domainLookupStart
-        const dnsEnd = entry.domainLookupEnd
-        const hasData = dnsEnd > 0 && dnsStart > 0
-        if (hasData) {
-          log.dnsTimeMs = Math.round(dnsEnd - dnsStart)
-          log.ttfbAfterDnsMs = Math.round(
-            entry.responseStart - entry.requestStart
-          )
-        }
-
-        if (log.httpProtocol === null && entry.nextHopProtocol) {
-          log.httpProtocol = entry.nextHopProtocol
-        }
-        // else response didn't have Timing-Allow-Origin: *
-        //
-        // if both dnsStart and dnsEnd are > 0 but have the same value,
-        // its a dns cache hit.
+  _getPerformanceMetricsForLog (log) {
+    const metrics = {}
+
+    const entry = performance
+      .getEntriesByType('resource')
+      .find((r) => r.name === log.url.href)
```

Comment:
can there be multiple entries for one url?

Reply:
There can but there's not much if anything to differentiate entries.  If we add something unique to the URL then that'll bust the browser cache which we don't want.


![image](https://github.com/filecoin-saturn/js-client/assets/4676043/9954ff39-179b-43b7-8463-316547c72fb7)

Entries contain a parsed version of the `server-timing` headers, so we could potentially match against that since we have access to the headers.

Reply:
Got it, I think this would be good to comment here. We know there can be multiple entries, we pick one because mostly we have no choice and it doesn't make a big difference